ArduCopter/mode_rtl.cpp

```diff
@@ -524,14 +538,26 @@ void ModeRTL::compute_return_target()
     target_alt = MAX(target_alt, MAX(g.rtl_altitude, RTL_ALT_MIN));
 
     // reduce climb if close to return target
-    float rtl_return_dist_cm = rtl_path.return_target.get_distance(rtl_path.origin_point) * 100.0f;
-    // don't allow really shallow slopes
+    // but don't allow really shallow slopes
     if (g.rtl_cone_slope >= RTL_MIN_CONE_SLOPE) {
-        target_alt = MAX(curr_alt, MIN(target_alt, MAX(rtl_return_dist_cm*g.rtl_cone_slope, curr_alt+RTL_ABS_MIN_CLIMB)));
+        target_alt = MAX(curr_alt, MIN(target_alt, MAX(rtl_cone_height, curr_alt+RTL_ABS_MIN_CLIMB)));
     }
 
-    // set returned target alt to new target_alt (don't change altitude type)
-    rtl_path.return_target.set_alt_cm(target_alt, (alt_type == ReturnTargetAltType::RELATIVE) ? Location::AltFrame::ABOVE_HOME : Location::AltFrame::ABOVE_TERRAIN);
+    Location::AltFrame target_alt_frame = Location::AltFrame::ABOVE_HOME;
+    switch (alt_type) {
```

Comment:
Maybe make this call a helper function....? 
`convert_ReturnTargetAltType_to_AltFrame(ReturnTargetAltType alt_type)` or just set the location in the Location altframe in the function too

The main function here is already pretty long.

Reply:
Done